lightning/src/routing/router.rs

```diff
@@ -2260,7 +2338,33 @@ where L::Target: Logger {
 	for results_vec in selected_paths {
 		let mut hops = Vec::with_capacity(results_vec.len());
 		for res in results_vec { hops.push(res?); }
-		paths.push(Path { hops, blinded_tail: None });
+
+		let blinded_path = payment_params.payee.blinded_route_hints().iter().enumerate()
+			.find(|(idx, (_, p))| {
+				DUMMY_BLINDED_SCID + *idx as u64 == hops[hops.len() - 1].short_channel_id ||
```

Comment:
I don't that would happen in practice because we factor segwit activation height into fake scid generation and we only run this for blinded route hints. But I agree it's iffy and caused some spurious test failures during development, so will update.

Reply:
Looks like we also can't use `DUMMY_BLINDED_SCID` in `used_channel_liquidities` for the same reason. Not sure what to do about that yet, but looking into adding a new variant for blinded recipients to `CandidateRouteHop`, so we can get rid off the dummy scid.